npm/test/clientprocessing.test.ts

```diff
@@ -0,0 +1,226 @@
+        fileId: 1,
+        checksum: "checksum123",
+        filePath: "file/path",
+        lastModified: 1,
+        size: 10
+      }
+    }
+    return { data }
+  }
+}
+
+class GraphqlClientFailureAddMetadata {
+  mutation: (
+    query: DocumentNode,
+    variables: TMockVariables
+  ) => Promise<FetchResult<IMockAddFileMetadata>> = async (
+    query: DocumentNode,
+    variables: TMockVariables
+  ) => {
+    return {
+      errors: [new GraphQLError("error 1"), new GraphQLError("error 2")]
+    }
+  }
+}
+
+class GraphqlClientFailureAddFiles {
+  mutation: (
+    query: DocumentNode,
+    variables: TMockVariables
+  ) => Promise<FetchResult<IMockAddFileData>> = async (
+    query: DocumentNode,
+    variables: TMockVariables
+  ) => {
+    const data: IMockAddFileData = null
+    return { data }
+  }
+}
+
+beforeEach(() => jest.resetModules())
+
+const mockKeycloak: KeycloakInstance<"native"> = {
+  init: jest.fn(),
+  login: jest.fn(),
+  logout: jest.fn(),
+  register: jest.fn(),
+  accountManagement: jest.fn(),
+  createLoginUrl: jest.fn(),
+  createLogoutUrl: jest.fn(),
+  createRegisterUrl: jest.fn(),
+  createAccountUrl: jest.fn(),
+  isTokenExpired: jest.fn(),
+  updateToken: jest.fn(),
+  clearToken: jest.fn(),
+  hasRealmRole: jest.fn(),
+  hasResourceRole: jest.fn(),
+  loadUserInfo: jest.fn(),
+  loadUserProfile: jest.fn(),
+  token: "fake-auth-token"
+}
+
+const mockSuccessAddFile: () => void = () => {
+  const mock = GraphqlClient as jest.Mock
+  mock.mockImplementation(() => {
+    return new GraphqlClientSuccessAddFiles()
+  })
+}
+
+const mockFailureAddFiles: () => void = () => {
+  const mock = GraphqlClient as jest.Mock
+  mock.mockImplementation(() => {
+    return new GraphqlClientFailureAddFiles()
+  })
+}
+
+const mockSuccessAddMetadata: () => void = () => {
+  const mock = GraphqlClient as jest.Mock
+  mock.mockImplementation(() => {
+    return new GraphqlClientSuccessAddMetadata()
+  })
+}
+
+const mockFailureAddMetadata: () => void = () => {
+  const mock = GraphqlClient as jest.Mock
+  mock.mockImplementation(() => {
+    return new GraphqlClientFailureAddMetadata()
+  })
+}
+
+test("processFiles returns list of file ids", async () => {
+  mockSuccessAddFile()
+  const client = new GraphqlClient("https://test.im", mockKeycloak)
+  const processing = new ClientFileProcessing(client)
+  const result = await processing.processFiles("1", 3)
+
+  expect(result).toEqual([1, 2, 3])
+})
+
+test("processFiles returns error if no data returned", async () => {
+  mockFailureAddFiles()
+  const client = new GraphqlClient("https://test.im", mockKeycloak)
+  const processing = new ClientFileProcessing(client)
+  await expect(processing.processFiles("1", 3)).rejects.toStrictEqual(
+    Error("Add files failed")
+  )
+})
+
+test("processClientFileMetadata adds extracted client file metadata to database", async () => {
+  const spyExtractFileMetadata = jest
+    .spyOn(mockFileInformation, "extractFileMetadata")
```

Comment:
Fairly minor thing, but does this have to be a spy? Can it be a stub instead?

Spies are often a bit of a code smell because they hint that there's a very fuzzy boundary between the code you're trying to test and its dependencies. If there's a clear boundary, you should be able to make it a stub or a mock. (Or not stub the implementation at all, which is the right choice in some cases, but might not make sense here if it's hard to construct realistic file objects.)